src/tezos_interop/long_lived_js_process.ml

```diff
@@ -61,30 +61,40 @@ module Pending : sig
   type t
   type add_error = private Duplicated_id of Id.t
   type resolve_error = private Unknown_id of Id.t
-
+  type kind =
+    | Listen
+    | Request
   val make : unit -> t
-  val add : t -> Id.t -> Yojson.Safe.t Lwt.u -> (unit, add_error) result
-  val resolve : t -> Id.t -> Yojson.Safe.t -> (unit, resolve_error) result
+  val add :
+    t -> Id.t -> kind -> (Yojson.Safe.t -> unit) -> (unit, add_error) result
+  val push : t -> Id.t -> Yojson.Safe.t -> (unit, resolve_error) result
 end = struct
-  type t = Yojson.Safe.t Lwt.u Id.Map.t ref
   type add_error = Duplicated_id of Id.t
   type resolve_error = Unknown_id of Id.t
+  type kind =
+    | Listen
+    | Request
+  type t = (kind * (Yojson.Safe.t -> unit)) Id.Map.t ref
 
   let make () = ref Id.Map.empty
   let find t id = Id.Map.find_opt id !t
   let mem t id = Id.Map.mem id !t
 
-  let add t id resolver =
+  let add t id kind resolver =
     if mem t id then
       Error (Duplicated_id id)
     else (
-      t := Id.Map.add id resolver !t;
+      t := Id.Map.add id (kind, resolver) !t;
       Ok ())
 
-  let resolve t id json =
+  let push t id json =
     match find t id with
-    | Some resolver ->
-      Lwt.wakeup_later resolver json;
+    | Some (Request, resolver) ->
+      t := Id.Map.remove id !t;
+      resolver json;
+      Ok ()
+    | Some (Listen, resolver) ->
+      resolver json;
```

Comment:
This normally removes something from the pending list, so I think it would be more appropriate to call it `Pending.pop` or something like that. I think `Pending.push` implies that it's adding something to the list of pending operations. (I didn't mind `Pending.resolve` either)

Reply:
Pending.resolve is also not correct as a Listen is not resolved, will be leaving the current name to not block on this.